code/modules/reagents/chemistry/reagents/food_reagents.dm

```diff
@@ -283,33 +283,33 @@
 	var/mob/living/carbon/victim = M
 	if(method == TOUCH || method == VAPOR)
 		//check for protection
-		var/mouth_covered = victim.is_mouth_covered()
 		var/eyes_covered = victim.is_eyes_covered()
 
 		//actually handle the pepperspray effects
-		if ( eyes_covered && mouth_covered )
+		if (eyes_covered && istype(victim.wear_mask, /obj/item/clothing/mask/gas))
 			return
-		else if ( mouth_covered )	// Reduced effects if partially protected
+		else if(istype(victim.wear_mask, /obj/item/clothing/mask/gas))	// Reduced effects if partially protected
 			if(prob(5))
 				victim.emote("scream")
-			victim.blur_eyes(3)
-			victim.blind_eyes(2)
-			victim.confused = max(M.confused, 3)
-			victim.damageoverlaytemp = 60
-			victim.Paralyze(60)
+			victim.blur_eyes(3) // 6 seconds
+			victim.blind_eyes(3) // 6 seconds
+			victim.confused = max(M.confused, 5)
 			return
 		else if ( eyes_covered ) // Eye cover is better than mouth cover
-			victim.blur_eyes(3)
-			victim.damageoverlaytemp = 30
+			victim.blur_eyes(2) // 4 seconds
+			victim.Knockdown(3 SECONDS)
+			victim.add_movespeed_modifier(MOVESPEED_ID_PEPPER_SPRAY, update=TRUE, priority=100, multiplicative_slowdown=0.25, blacklisted_movetypes=(FLYING|FLOATING))
+			addtimer(CALLBACK(victim, /mob.proc/remove_movespeed_modifier, MOVESPEED_ID_PEPPER_SPRAY), 10 SECONDS)
 			return
```

Comment:
I'm usually all for returning as soon as possible but you can get rid of three lines of duplicate code if you don't.
```
			victim.Knockdown(3 SECONDS)
			victim.add_movespeed_modifier(MOVESPEED_ID_PEPPER_SPRAY, update=TRUE, priority=100, multiplicative_slowdown=0.25, blacklisted_movetypes=(FLYING|FLOATING))
			addtimer(CALLBACK(victim, /mob.proc/remove_movespeed_modifier, MOVESPEED_ID_PEPPER_SPRAY), 10 SECONDS)
```
are shared behaviour between this case and the next.

Reply:
Ah, I was combining the effects of the two.